packages/cli/src/LoadNodesAndCredentials.ts

```diff
@@ -92,6 +90,12 @@ export class LoadNodesAndCredentials implements INodesAndCredentials {
 		await this.loadNodesFromCustomDirectories();
 		await this.postProcessLoaders();
 		this.injectCustomApiCallOptions();
+
+		this.on('postProcess:start', async () => {
+			await this.postProcessLoaders();
+			await this.generateTypesForFrontend();
+			this.emit('postProcess:done');
```

Comment:
I think this is working as expected because `EventEmitter` is synchronous. But, the correct order should be to generate frontend types **after** the post processing is done.

I'll update this class to have an array of post-processing functions, and update `NodeTypes` to add `applySpecialNodeParameters` to that array. That way we can simplify the code here, and also guarantee that frontend types are always correct.

The code flow with the events is hard to follow.

Reply:
also, `injectCustomApiCallOptions` needs to be in that array, to make sure that even community nodes get processed by that function.